AdvancedTaskMarsPart2/StepDefinitions/CertificationsFeatureStepDefinition.cs

```diff
@@ -0,0 +1,78 @@
+﻿using AdvancedTaskMarsPart2.AssertHelpers;
+using AdvancedTaskMarsPart2.Model;
+using AdvancedTaskMarsPart2.Pages.Components.ProfileOverview;
+using AdvancedTaskMarsPart2.Pages.Components.SignIn;
+using AdvancedTaskMarsPart2.TestData;
+using AdvancedTaskMarsPart2.Utilities;
+using TechTalk.SpecFlow;
+
+namespace AdvancedTaskMarsPart2.StepDefinitions
+{
+    [Binding]
+    public class CertificationsFeatureStepDefinition : CommonDriver
+    {
+        SignInComponent signInComponent;
+        LoginInComponent loginInComponent;
+        ProfileMenuTabsComponents profileMenuTabsComponents;
+        AddAndDeleteCertificationsComponent addAndDeleteCertificationsComponent;
+        ProfileCertificationsOverviewComponent profileCertificationsOverviewComponent;
+
+        public CertificationsFeatureStepDefinition()
+        {
+            signInComponent = new SignInComponent();
+            loginInComponent = new LoginInComponent();
+            profileMenuTabsComponents = new ProfileMenuTabsComponents();
+            addAndDeleteCertificationsComponent = new AddAndDeleteCertificationsComponent();
+            profileCertificationsOverviewComponent = new ProfileCertificationsOverviewComponent();
+        }
+
+        [Given(@"User logged into Mars URL and navigates to Certifications tab")]
+        public void GivenUserLoggedIntoMarsURLAndNavigatesToCertificationsTab()
+        {
+            signInComponent.clickSignInButton();
+            List<UserInformation> userInformatioList = JsonReader.LoadData<UserInformation>(@"UserInformation.json");
+            foreach (var userInformation in userInformatioList)
+            {
+                loginInComponent.LoginActions(userInformation);
+            }
+            profileMenuTabsComponents.clickCertificationsTab();
+        }
+
+        [When(@"Delete all records in the certifications list")]
+        public void WhenDeleteAllRecordsInTheCertificationsList()
+        {
+            addAndDeleteCertificationsComponent.DeleteAllRecords();
```

Comment:
so i see that you have thought about state but i want you to think about cleanup as well 

so think about first clean up before any test runs (you delete all data around this feature e.g  before any language test run you delete all languages). Then after each test delete only data that the test has added! (test1 added english and test2 to added french then test1 should only delete english and test2 should delete french only) 

also just a note around state say you are updating a language then you must added language then update it do not assume the language is there!